### PR TITLE
sys/net/gnrc: provide lora adaption to netif

### PR DESCRIPTION
**Pull request by aabadie**
Provide LoRa adaption to the new netif interface.

Follow-up of #7950.

### CODE REVIEW

File: sys/net/gnrc/netif/gnrc_netif_lora.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    payload = gnrc_pktbuf_add(NULL, NULL, pktlen, GNRC_NETTYPE_UNDEF);
+    if (payload == NULL) {
+        DEBUG("[lora-gnrc] recv: unable to allocate space in the pktbuf\n");
+        /* tell the driver to drop the packet */
+        dev->driver->recv(dev, NULL, 1, NULL);
+        return NULL;
+    }
+
+    /* copy the complete payload into the packet buffer */
+    dev->driver->recv(dev, payload->data, pktlen, NULL);
```
Comment (miri64) [inline sys/net/gnrc/netif/gnrc_netif_lora.c]: Doesn't LoRA has any L2 headers oO (if so, I guess you *can* use `gnrc_netif2_raw` after all)?

Comment (miri64) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: (but from what I have seen on some websites, there is a LoRA MHDR)

Comment (aabadie) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: LoRa (as used here) is just radio data. LoRaMac has L2 headers.

Comment (miri64) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: Ok, now I need some clarification: Where in the LoRa-Stack is this here to go? Is this just some proof-of-concept that will be insignificant as soon as the real LoRaMAC `gnrc_netif` layer comes? Is this supposed to go below LoRaMac (which would be possible with GNRC, but why have the extra thread then)? In either case: I am struggling to see the long-term benefit of this PR, but maybe I'm missing something.

Comment (aabadie) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: > Is this just some proof-of-concept that will be insignificant as soon as the real LoRaMAC gnrc_netif layer comes?

I was more thinking of another netif type for loramac.

> In either case: I am struggling to see the long-term benefit of this PR, but maybe I'm missing something.

Or maybe I am missing something (most likely I would say ;) )

**Comment by aabadie** [comment]
I think I should move all this in the sx127x driver code instead.and rename the netif create in `gnrc_netif_sx127x_create`. @miri64 what do you think ?

**Comment by aabadie** [comment]
What I want to do is auto initializing the radio driver and being able to send receive packets using `ifconfig` (remember #7482). Here there's no LoRamac stuff. I'm planning to add loramac using an external package that has all the functions to parse the headers and manage the timings see [this WIP branch](https://github.com/aabadie/RIOT/tree/pkg_semtech-loramac).

Maybe in both cases (raw radio packet or packet via semtech-loramac/rn2xx3 mac) the gnrc_netif_raw could be used ?

**Comment by miri64** [comment]
> What I want to do is auto initializing the radio driver and being able to send receive packets using ifconfig (remember #7482). 

ifconfig (as the name suggests) can only configure an interface...

> Here there's no LoRamac stuff. I'm planning to add loramac using an external package that has all the functions to parse the headers and manage the timings see this WIP branch.
>
> Maybe in both cases (raw radio packet or packet via semtech-loramac/rn2xx3 mac) the gnrc_netif_raw could be used ?

Yes.

**Comment by aabadie** [comment]
> ifconfig (as the name suggests) can only configure an interface...

indeed, I was more thinking at `txtsnd` for sending and `set` subcommand of `ifconfig` for putting the radio in listen mode.

File: sys/net/gnrc/netif/gnrc_netif_lora.c (hunk 1 above)
Comment (PeterKietzmann) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: @aabadie wouldn't `gnrc_netif_raw` fit your needs?

Comment (PeterKietzmann) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: You actually used [`gnrc_netif_raw`](https://github.com/RIOT-OS/RIOT/pull/8159/files#diff-45bbe6e7306e5287cbc78b86d405120aR59) in #8159 right?

Comment (aabadie) [reply sys/net/gnrc/netif/gnrc_netif_lora.c]: @PeterKietzmann, yes, so I think we can close this one.

**Comment by miri64** [comment]
@aabadie wrote

> @PeterKietzmann, yes, so I think we can close this one.

**Pull request closed without merging**